src/react-imgix.js

```diff
@@ -63,43 +39,89 @@ export default class ReactImgix extends Component {
     disableLibraryParam: PropTypes.bool
   };
   static defaultProps = {
-    aggressiveLoad: false,
     auto: ["format"],
     entropy: false,
     faces: true,
     fit: "crop",
-    fluid: true,
-    generateSrcSet: true,
+    disableSrcSet: false,
     onMounted: () => {},
-    precision: 100,
     type: "img"
   };
-  state = {
-    width: null,
-    height: null,
-    mounted: false
-  };
 
-  forceLayout = () => {
+  componentDidMount = () => {
     const node = ReactDOM.findDOMNode(this);
-    this.setState({
-      width: node.scrollWidth,
-      height: node.scrollHeight,
-      mounted: true
-    });
     this.props.onMounted(node);
   };
 
-  componentDidMount = () => {
-    this.forceLayout();
-  };
+  buildSrcs = () => {
+    const props = this.props;
+    const {
+      width,
+      height,
+      entropy,
+      faces,
+      auto,
+      customParams,
+      disableLibraryParam,
+      disableSrcSet,
+      type
+    } = props;
+
+    let crop = false;
+    if (faces) crop = "faces";
+    if (entropy) crop = "entropy";
+    if (props.crop) crop = props.crop;
+
+    let fit = false;
+    if (entropy) fit = "crop";
```

Comment:
I think this should be `if (entropy || faces)`?

Reply:
Updated